python/semantic_kernel/connectors/ai/open_ai/services/open_ai_chat_completion_base.py

```diff
@@ -269,9 +276,25 @@ def _chat_message_content_to_dict(self, message: "ChatMessageContent") -> dict[s
 
     async def _send_chat_request(self, settings: OpenAIChatPromptExecutionSettings) -> list["ChatMessageContent"]:
         """Send the chat request."""
+        span = model_diagnostics.start_completion_activity(settings.ai_model_id, MODEL_PROVIDER_NAME,
+                                                           self._settings_messages_to_prompt(settings.messages),
+                                                           settings)
+
         response = await self._send_request(request_settings=settings)
         response_metadata = self._get_metadata_from_chat_response(response)
-        return [self._create_chat_message_content(response, choice, response_metadata) for choice in response.choices]
+        
+        chat_message_contents = [self._create_chat_message_content(response, choice, response_metadata)
+                                  for choice in response.choices]
+        
+        if span is not None:
+            finish_reasons: list[str] = []
+            for choice in response.choices:
+                finish_reasons.append(choice.finish_reason)
+            with trace.use_span(span, end_on_exit=True):
+                model_diagnostics.set_completion_response(span, chat_message_contents, finish_reasons, response.id,
+                                                          response.usage.prompt_tokens,
+                                                          response.usage.completion_tokens)
```

Comment:
nit: is this auto-formatted?

Reply:
I just installed the Ruff extension is VS Code and now the formatting is automatic and probably more like what you expected

Reply:
Totally.  Changed.